Core-perf: Netty HttpClient and Ok HttpClient performance (#27982)
Azure/azure-sdk-for-java: sdk/core/azure-core-http-netty/CHANGELOG.md

## 1.12.0-beta.1 (Unreleased)

### Features Added

> - The `NettyAsyncHttpClientProvider.createInstance()` now have option to share one single shared `HttpClient`.

Reviewer: ```suggestion
- The `NettyAsyncHttpClientProvider.createInstance()` now has the option to share a shared `HttpClient`.
```

>   A new configuration environment variable `AZURE_ENABLE_HTTP_CLIENT_SHARING` has been added for backwards
>   compatibility. Setting it to `true` before starting the process to use single shared `HttpClient`.

Reviewer: Since the flag has changed from opt-out to opt-in this statement can be changed.

```suggestion
  Set the environment variable `AZURE_ENABLE_HTTP_CLIENT_SHARING`  to `true` before starting the process to use the shared `HttpClient`.
```

## 1.12.0-beta.1 (Unreleased)

### Features Added
  
### Breaking Changes
- The `NettyAsyncHttpClientProvider.createInstance()` now returns shared instance of `HttpClient`. Azure libraries that 
  don't configure custom `HttpClientOptions` or `HttpClient` instance are now using single shared instance as well. 

>   A new configuration environment variable `AZURE_DISABLE_HTTP_CLIENT_SHARING` has been added for backwards 

Reviewer 2: Doesn't it default to true if unset? So the normal configuration is to use previous behavior, and the only way to get new behavior is to set it to `false`?

Author: Yes. Was thinking of not breaking existing customers' behavior. So setting it to 'true' will make no change to the customer who running the code without break.

Reviewer 2: Agreed we should default to true to maintain behavior. I think the doc here is confusing as it says to do the thing that happens automatically. I would state that setting to false enables new behavior, and leaving it unset or setting it to true maintains existing behavior.

Author: updated

>   compatibility setting it to `true` before starting the process brings back previous behavior.

Reviewer 2: ```suggestion
  compatibility. Setting it to `true` before starting the process brings back previous behavior.
```
